tests/e2e_test.go

```diff
@@ -63,11 +73,31 @@ var _ = ginkgo.BeforeSuite(func() {
 		teleporterContractAddress,
 		fundedKey,
 	)
+
+	var ctx context.Context
+	ctx, cancelDecider = context.WithCancel(context.Background())
+	// we'll call cancelDecider in AfterSuite, but also call it if this
+	// process is killed, because AfterSuite won't always run then:
+	signalChan := make(chan os.Signal, 2)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-signalChan
+		cancelDecider()
+	}()
+	decider = exec.CommandContext(ctx, "./tests/cmd/decider/decider")
+	decider.Start()
+	log.Info("Started decider service")
+
```

Comment:
I don't think this is necessary. If ginkgo exits for whatever reason, everything that it was running should be cleaned up - we don't currently look for OS signals for the other processes we run.

What we _do_ need is something that watches the decider process, and if it exists abnormally, we should panic. Example here https://github.com/ava-labs/awm-relayer/blob/main/tests/utils/utils.go#L91

Reply:
Are you sure it's not necessary? A little while ago I tried interrupting a test with Ctrl+C, and from the stdout it didn't look like it ran the `AfterSuite` hook, and the `decider` process was still running. And just now I tried again and saw that it _did_ execute that hook, and the decider process was _not_ running. So I think the behavior is inconsistent.

If the decider is left running/orphaned/defunct after the tests have finished, then it will fail to start in the next test run, because the port it uses is always the same and that port will be in use if it's already running from a previous test run. And if you're working on some changes totally unrelated to the decider, it could be pretty confusing and unclear what's going on.

This uncertainty and fear of confusion is why I decided to put this code into place. I'm happy to defer to the team's opinion though.

The idea of watching the decider for abnormal exit is a great idea, thanks, I'll do that.

Reply:
Yeah, I think we should move to running our e2e tests in a docker container (if feasible), that way we get the cleanup for free, and wouldn't need to check for OS signals, but until then, this is helpful.